quickcomm/admin.py

```diff
@@ -1,13 +1,148 @@
+    def profile_image_html(self, obj: Author):
+        if obj is None or obj.profile_image is None:
+            return format(f"<p>No image found</p>")
+
+        return format_html(f"<img src={obj.profile_image} width=100pt height=100pt />")
+
+
+    @admin.display(description='Location of Author')
+    def location(self, obj: Author):
+        return obj.location
+
+
+    # @admin.action(description='Sync posts')
+    # def sync_posts(self, request, queryset):
+    #     for author in queryset:
+    #         author.sync_posts()
+    #     self.message_user(request, f"Synced posts for {queryset.count()} authors")
+
+    def has_add_permission(self, request, obj=None):
+        return True
+
+    def has_change_permission(self, request, obj: Author=None):
+        if obj is None:
+            return True
+
+        if obj.is_local:
+            return True
+
+        return False
+
+
+class AuthorInlineForHost(admin.TabularInline):
+    model = Author
+    extra = 0
+    fields = ['display_name', 'external_url']
+    show_change_link = True
+    can_delete = False
+
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    @admin.display(description='Display Name')
+    def display_name(self, obj: Author):
+        return obj.display_name
+
+@admin.register(Host)
+class HostAdmin(admin.ModelAdmin):
+    list_display = ('nickname_or_url', 'url', 'last_ping_result')
+    readonly_fields = ('last_successful_ping', 'last_ping', 'last_ping_result')
+    actions_on_top = True
+    actions = ['ping']
+    inlines = [AuthorInlineForHost]
+
+    change_form_template = 'admin/edit_host.html'
+
+    fieldsets = (
+        (None, {
+            'description': 'A host is a remote server that follows the CMPUT 404 Social Distribution protocol.',
+            'fields': ('url', 'nickname')
+        }),
+        ('Ping Details', {
+            'classes': ('collapse'),
+            'description': 'These fields are automatically updated when the host is pinged. The host is pinged occasionally in the background to check if it is still online, and when it is added or edited.',
+            'fields': ('last_successful_ping', 'last_ping', 'last_ping_result')
+        }),
+    )
+
+    @admin.display(description='Host')
+    def nickname_or_url(self, obj: Host):
+        return obj.nickname_or_url
+
+    @admin.action(description='Ping')
+    def ping(self, request, queryset):
+        for host in queryset:
+            host.ping()
+        # save the hosts
+        queryset.update()
+        self.message_user(request, f"Pinged {queryset.count()} hosts")
+
+    @admin.action(description='Sync author details')
+    def sync_author_details(self, request, queryset):
+        for host in queryset:
+            host.sync_authors()
+
+    # Taken from https://books.agiliq.com/projects/django-admin-cookbook/en/latest/custom_button.html
```

Comment:
do we add more to inline citation (i.e. date accesssed, author, title of article) or just in README?